### PR TITLE
test: add 'integration-test' to e2e runs

### PR DESCRIPTION
**Pull request by smira**
Also refactored `integration-test` build as a generic step to be shared
by basic-integration and e2e-integration steps.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Comment by smira** [comment]
what is the easiest way to trigger e2e to make it run? just add to the basic flow temporarily?

File: hack/test/e2e-integration.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ e2e_run "timeout=\$((\$(date +%s) + ${TIMEOUT}))
          until [ -n \"\${STATUS_TALOSCONFIG}\" ]; do
            [[ \$(date +%s) -gt \$timeout ]] && exit 1
            sleep 10
-           STATUS_TALOSCONFIG=\$( KUBECONFIG=${TMP}/kubeconfig kubectl get talosconfig ${NAME_PREFIX}-controlplane-0 -o jsonpath='{.status.talosConfig}' )
```
Comment (andrewrynhard) [inline hack/test/e2e-integration.sh]: ❤️

**Comment by andrewrynhard** [comment]
> what is the easiest way to trigger e2e to make it run? just add to the basic flow temporarily?

You will need the drone cli:

```
drone build promote talos-systems/talos 4161 e2e
```

**Comment by smira** [comment]
There's a clean e2e run now: https://ci.dev.talos-systems.io/talos-systems/talos/4889/1/31

(with bumped e2e k8s version)

**Pull request merged**